pkg/utils/jsonutils.go

```diff
@@ -58,3 +63,61 @@ func FlattenSingleValue(key string, m map[string]interface{}, child interface{})
 		m[key] = child
 	}
 }
+
+func ExtractJsonValueBuffered(data []byte, workBuf []byte, keys ...string) ([]byte, jsonparser.ValueType, error) {
+	var err error
+	decoder := json.NewDecoder(bytes.NewReader(data))
+	decoder.UseNumber()
+	var current interface{} = nil
+	for _, key := range keys {
+		if err = decoder.Decode(&current); err != nil {
+			return nil, jsonparser.NotExist, fmt.Errorf("error decoding JSON: %v", err)
+		}
+		switch obj := current.(type) {
+		case map[string]interface{}:
+			if value, found := obj[key]; found {
+				current = value
+			} else {
+				return nil, jsonparser.NotExist, fmt.Errorf("key %s not found", key)
+			}
+		case []interface{}:
+			return nil, jsonparser.NotExist, fmt.Errorf("unexpected array while traversing key %s", key)
+		default:
+			return nil, jsonparser.NotExist, fmt.Errorf("expected object for key %s but found %T", key, obj)
+		}
+	}
+	valueBytes, err := json.Marshal(current)
+	if err != nil {
+		return nil, jsonparser.NotExist, fmt.Errorf("error converting value to bytes: %v", err)
+	}
+	workBuf = ResizeSlice(workBuf, len(valueBytes))
+	copy(workBuf, valueBytes)
```

Comment:
You're still not using the workbuf properly (i.e., even if the user passes a workbuf large enough, you'll still allocate memory and then copy it to the workbuf)